src/domain/sales-channels/tables/product.tsx

```diff
@@ -0,0 +1,462 @@
+import Table, { TablePagination } from "../../../components/molecules/table"
+import { SALES_CHANNEL_PRODUCTS_TABLE_COLUMNS } from "./config"
+import useQueryFilters from "../../../hooks/use-query-filters"
+import { useProductFilters } from "../../../components/templates/product-table/use-filter-tabs"
+
+/* ****************************************** */
+/* ************** TABLE CONFIG ************** */
+/* ****************************************** */
+
+const DEFAULT_PAGE_SIZE = 12
+
+/**
+ * Default filtering config for querying products endpoint.
+ */
+const defaultQueryProps = {
+  additionalFilters: {
+    expand: "collection,sales_channels",
+    fields: "id,title,type,thumbnail,status",
+  },
+  limit: DEFAULT_PAGE_SIZE,
+  offset: 0,
+}
+
+/* ******************************************** */
+/* ************** PRODUCTS TABLE ************** */
+/* ******************************************** */
+
+type ProductTableProps = {
+  isAddTable: boolean
+  count: number
+  products: Product[]
+  setSelectedRowIds: (ids: string[]) => void
+  selectedRowIds: string[]
+  removeProductFromSalesChannel: (ids: string[]) => void
+}
+
+/**
+ * Renders a table of sales channel products.
+ */
+function ProductTable(props: ProductTableProps) {
+  const {
+    productFilters: {
+      setTab,
+      saveTab,
+      removeTab,
+      availableTabs: filterTabs,
+      activeFilterTab,
+      setFilters,
+      reset,
+      filters,
+    },
+    paginate,
+    setQuery: setFreeText,
+    queryObject,
+
+    // CONTAINER props
+    isAddTable,
+    count,
+    products,
+    setSelectedRowIds,
+    removeProductFromSalesChannel,
+  } = props
+
+  const offs = parseInt(queryObject.offset) || 0
+  const limit = parseInt(queryObject.limit)
+
+  const [query, setQuery] = useState(queryObject.query)
+  const [numPages, setNumPages] = useState(0)
+
+  const clearFilters = () => {
+    reset()
+    setQuery("")
+  }
+
+  useEffect(() => {
+    if (typeof count !== "undefined") {
+      const controlledPageCount = Math.ceil(count / limit)
+      setNumPages(controlledPageCount)
+    }
+  }, [count])
+
+  const {
+    getTableProps,
+    getTableBodyProps,
+    headerGroups,
+    rows,
+    prepareRow,
+    gotoPage,
+    canPreviousPage,
+    canNextPage,
+    pageCount,
+    nextPage,
+    previousPage,
+    // Get the state from the instance
+    state: { pageIndex, pageSize, ...state },
+  } = useTable(
+    {
+      columns: SALES_CHANNEL_PRODUCTS_TABLE_COLUMNS,
+      data: products || [],
+      manualPagination: true,
+      initialState: {
+        pageIndex: Math.floor(offs / limit),
+        pageSize: limit,
+      },
+      pageCount: numPages,
+      autoResetPage: false,
+      autoResetSelectedRows: false,
+      getRowId: (row) => row.id,
+    },
+    usePagination,
+    useRowSelect
+  )
+
+  useEffect(() => {
+    setSelectedRowIds(Object.keys(state.selectedRowIds))
+  }, [Object.keys(state.selectedRowIds).length])
```

Comment:
Issue: This doesn't show the already selected products in the product modal

Reply:
but there are no initially selected products in the modal rn

Reply:
I mean that if I have added products to the sales channel, those are not indicated in the current modal when i open it to add products

Reply:
this depends on the discussion below, I will make the change depending on the conclusion

Reply:
Selected products are not indicated as selected or removed from the list in the available products modal. If we go with the one page modal setup as you suggested we should remove already added products and update the title of the actionable.